CruCentralCoast/Events/EventsVC.swift

```diff
@@ -48,47 +57,102 @@ class EventsTVC: UITableViewController {
         longPressGestureRecognizer.delegate = self
         self.tableView.addGestureRecognizer(longPressGestureRecognizer)
     }
     
     @objc internal func handleLongPressGesture(gestureRecognizer: UILongPressGestureRecognizer) {
         let touchPoint = gestureRecognizer.location(in: self.view)
         if gestureRecognizer.state == .began {
+            self.longPressCancelled = false
+            self.longPressFailed = false
             if let indexPath = tableView.indexPathForRow(at: touchPoint) {
-                let cell = tableView.cellForRow(at: indexPath)
-                UIView.animate(withDuration: 0.2, animations: {
-                    cell?.transform = CGAffineTransform(scaleX: 0.9, y: 0.9)
-                })
+                self.longPressIndexPath = indexPath
+                self.longPressGestureBegan(indexPath: indexPath)
             }
         } else if gestureRecognizer.state == .ended {
             if let indexPath = tableView.indexPathForRow(at: touchPoint) {
-                let cell = tableView.cellForRow(at: indexPath)
-                guard let vc = UIStoryboard(name: "Events", bundle: nil).instantiateViewController(withIdentifier: "EventDetailsVC") as? EventDetailsVC else {
-                    assertionFailure("Probably used the wrong storyboard name or identifier here")
-                    return
+                if indexPath == self.longPressIndexPath {
+                    self.longPressGestureEnded(indexPath: indexPath)
+                } else {
+                    self.longPressGestureCancelled(indexPath: self.longPressIndexPath)
                 }
-                UIView.animate(withDuration: 0.05, animations: {
-                    cell?.transform = CGAffineTransform.identity
-                }, completion: { (finish) in
-                    vc.configure(with: self.dataArray[indexPath.item])
-                    vc.transitioningDelegate = self
-                    self.selectedCell = self.tableView.cellForRow(at: indexPath) as? EventsTableCell
-                    self.navigationController?.present(vc, animated: true, completion: nil)
-                })
             }
         } else if gestureRecognizer.state == .cancelled {
+            self.longPressCancelled = true
             if let indexPath = tableView.indexPathForRow(at: touchPoint) {
-                let cell = tableView.cellForRow(at: indexPath)
-                UIView.animate(withDuration: 0.2, animations: {
-                    cell?.transform = CGAffineTransform.identity
-                })
+                self.longPressGestureCancelled(indexPath: indexPath)
+            }
+        }
+        else if gestureRecognizer.state == .failed {
+            self.longPressFailed = true
+            if let indexPath = tableView.indexPathForRow(at: touchPoint) {
+                self.longPressGestureCancelled(indexPath: indexPath)
             }
         }
     }
+    
+    func longPressGestureBegan(indexPath: IndexPath) {
```

Comment:
It would be great to consolidate all of these gesture methods in an extension of the class `extension EventsVC: UIGestureRecognizerDelegate`